ykfr/src/lib.rs

```diff
@@ -303,22 +294,10 @@ impl FrameReconstructor {
                         // Direct locations are always be in regards to RBP.
                         assert_eq!(*reg, 6);
                         let temp = unsafe { rbp.offset(isize::try_from(*off).unwrap()) };
-                        match size {
-                            4 => {
-                                let dval = unsafe { ptr::read(val as *mut u32) };
-                                unsafe { ptr::write(temp as *mut u32, dval as u32) };
-                            }
-                            8 => {
-                                let dval = unsafe { ptr::read(val as *mut u64) };
-                                unsafe { ptr::write(temp as *mut u64, dval) };
-                            }
-                            _ => {
-                                todo!()
-                            }
-                        }
+                        unsafe { libc::memcpy(temp, val as *const c_void, size as usize) };
```

Comment:
`as` in Rust is dangerous, so we should always add a (brief) comment explaining why it's safe. In this case I'm not sure what type `size` has before the cast.

Reply:
`size` is a `u64` (`c_ulonglong`). But I've changed this to use `try_from` in [607e912](https://github.com/ykjit/yk/pull/589/commits/607e9126124fbc543121b222fa75ad0899bc212f)